tools/test/TestCSVFileImport.cc

```diff
@@ -90,3 +90,27 @@ TEST (TestCSVFileImport, testTimezoneOption) {
     EXPECT_EQ("", error);
   }
 }
+
+TEST (TestCSVFileImport, testLongBinary) {
+  // create an ORC file from importing the CSV file
+  const std::string pgm1 = findProgram("tools/src/csv-import");
+  const std::string csvFile = findExample("TestCSVFileImport.testLongBinary.csv");
```

Comment:
I think this test does not reveal the bug. We need a test that would fail without the fix. So I think we need a csv file with strings larger than 4MB. It'd be better to create the csv file in the test instead of importing it staticly in this project.

Reply:
Yes, the csv file is not covering this case. Sorry I am not familiar with gtest enough, is there any recommended way doing this, like generating data on the fly?

Reply:
I added the the csv file creation in TestCSVFileImport.cc to create temp file in the test